var/spack/repos/builtin/packages/flex/package.py

```diff
@@ -43,6 +43,10 @@ class Flex(AutotoolsPackage):
     # - https://github.com/westes/flex/issues/241
     patch('https://github.com/westes/flex/commit/24fd0551333e7eded87b64dd36062da3df2f6380.patch', sha256='09c22e5c6fef327d3e48eb23f0d610dcd3a35ab9207f12e0f875701c677978d3', when='@2.6.4')
 
+    def autoreconf(self, spec, prefix):
```

Comment:
I think this will break 2.6.3 and 2.6.1. We only add the Autotools dependencies for 2.6.4 and :2.6.0, so autoreconf won't work for these other versions. Maybe you could combine the `force_autoreconf` logic into here, or add the dependency for every version? Alternatively, we could try to write a patch for `configure` that doesn't require autoreconf.

Reply:
so as soon as an `autoreconf` method is defined in an `Autotools` package it will always reconfigure?

Reply:
Looks like it. :confused: Adding `@when('@:2.6.0,2.6.4')` fixed it for me.

Reply:
The failure I experienced yesterday has gone away with this fix.